content/src/denylist/Denylist.ts

```diff
@@ -26,9 +26,12 @@ export class Denylist {
     return `${actionMessage}-${target.asString()}-${timestamp}`
   }
 
-  async addTarget(target: DenylistTarget, metadata: DenylistMetadata) {
+  async addTarget(target: DenylistTarget, metadata: DenylistMetadata): Promise<OperationResult> {
     // Validate blocker and signature
-    await this.validateSignature(DenylistAction.ADDITION, target, metadata)
+    const operationResult: OperationResult = await this.validateSignature(DenylistAction.ADDITION, target, metadata)
+    if (isSuccessfulOperation(operationResult)) {
```

Comment:
What? If the signature validation is successful then you should continue and write everything to the database